internal/k8s/controllers/service_controller_reload.go

```diff
@@ -0,0 +1,103 @@
+/*
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+
+	"github.com/go-kit/kit/log/level"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+)
+
+type reloadEvent struct {
+	metav1.TypeMeta
+	metav1.ObjectMeta
+}
+
+func (evt *reloadEvent) DeepCopyObject() runtime.Object {
+	res := new(reloadEvent)
+	res.Name = evt.Name
+	res.Namespace = evt.Namespace
+	return res
+}
+
+func NewReloadEvent() event.GenericEvent {
+	evt := reloadEvent{}
+	evt.Name = "reload"
+	evt.Namespace = "metallbreload"
+	return event.GenericEvent{Object: &evt}
+}
+
+func isReloadReq(req ctrl.Request) bool {
+	if req.Name == "reload" && req.Namespace == "metallbreload" {
+		return true
+	}
+	return false
+}
+
+func (r *ServiceReconciler) reprocessAll(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	level.Info(r.Logger).Log("controller", "ServiceReconciler - reprocessAll", "start reconcile", req.NamespacedName.String())
+	defer level.Info(r.Logger).Log("controller", "ServiceReconciler - reprocessAll", "end reconcile", req.NamespacedName.String())
+
+	var services v1.ServiceList
+	if err := r.List(ctx, &services); err != nil {
+		level.Error(r.Logger).Log("controller", "ServiceReconciler - reprocessAll", "error", "failed to list the services", "error", err)
```

Comment:
"error" is used twice, i think we said it's only showing the second one this way